tools/protoprint/protoprint.py

```diff
@@ -726,20 +740,29 @@ def visit_file(self, file_proto, type_context, services, msgs, enums):
         return clang_format(header + formatted_services + formatted_enums + formatted_msgs)
 
 
-if __name__ == '__main__':
-    proto_desc_path = sys.argv[1]
+class ProtoprintTraverser:
+
+    def traverse_file(self, file_proto, visitor):
+        # This breaks the Visitor contract, ie class props should not be set on individual visitation,
+        # but, its already broken, so this just works around for now.
```

Comment:
i think this is correct - it confused me at first as i didnt expect any instance props to be set - but adding this fixed the problem

Reply:
perhaps it can be passed in to the visitor methods by the traverser